javascript/ql/src/semmle/javascript/dataflow/Nodes.qll

```diff
@@ -125,7 +125,20 @@ class InvokeNode extends DataFlow::SourceNode {
    * This predicate can be overridden to alter the call graph used by the interprocedural
    * data flow libraries.
    */
-  Function getACallee(int imprecision) { result = InvokeNode::getACallee(this, imprecision) }
+  cached
+  Function getACallee(int imprecision) {
+    result.flow() = getCalleeNode().getAFunctionValue(imprecision)
+    or
+    imprecision = 0 and
+    exists(InvokeExpr expr | expr = this.(DataFlow::Impl::ExplicitInvokeNode).asExpr() |
+      result = expr.getResolvedCallee()
```

Comment:
IIUC, we previously only used this if we weren't able to find a callee using data flow. Can you remind me why we dropped that side condition?

Reply:
From the PR description:

> Previously, the call graph exposed through getACallee(0) had the nice property that if any callees were found, we almost certainly had all of them. This is doesn't quite hold anymore. Since it now includes results from type tracking, there will be cases where a proper subset of the callees could be found (where previously nothing was found). This has a few ramifications:

> Static types now contribute to the call graph even if some edges were also found through data flow. We would lose useful call edges otherwise.

The negative dependency meant that adding more data flow could result in fewer call edges.